misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/natives/mock/ObjectMock.java

```diff
@@ -267,21 +540,43 @@ public static BError thenReturn(BObject caseObj) {
         }
         ObjectType objectType = (ObjectType) TypeUtils.getImpliedType(genericMock.getType());
         if (functionName != null) {
+
             // register return value for member function
-            BArray args = caseObj.getArrayValue(StringUtils.fromString("args"));
-            if (!validateReturnValue(functionName, returnVal, objectType.getMethods())) {
-                String detail =
-                        "return value provided does not match the return type of function '" + functionName + "()'";
-                return ErrorCreator.createError(
-                        MockConstants.TEST_PACKAGE_ID,
-                        MockConstants.FUNCTION_SIGNATURE_MISMATCH_ERROR,
-                        StringUtils.fromString(detail),
-                        null,
-                        new MapValueImpl<>(PredefinedTypes.TYPE_ERROR_DETAIL));
+            BArray args = caseObj.getArrayValue(StringUtils.fromString(MockConstants.ARGS));
+            if (MockConstants.MEMBER_RESOURCE_FUNCTION_STUB.equals(mockType)) {
+                BArray pathArgs = caseObj.getArrayValue(StringUtils.fromString(MockConstants.PATH_ARGS));
+                String functionPattern = getFunctionNameForResourcePath(functionName);
+                String accessor = caseObj.getStringValue(StringUtils.fromString(MockConstants.ACCESSOR)).toString();
+                String resourceFunctionPattern = MockConstants.DOLLAR_RESOURCE_SEPARATOR + accessor +
+                        MockConstants.DOLLAR_RESOURCE_SEPARATOR + functionPattern;
+                if (!validateReturnValueForResourcePath(resourceFunctionPattern, returnVal,
+                        ((BClientType) objectType).getResourceMethods())) {
+                    String detail =
+                            "return value provided does not match the return type of the resource path '" +
+                                    functionName + "'";
+                    return ErrorCreator.createError(
+                            MockConstants.TEST_PACKAGE_ID,
+                            MockConstants.FUNCTION_SIGNATURE_MISMATCH_ERROR,
+                            StringUtils.fromString(detail),
+                            null,
+                            new MapValueImpl<>(PredefinedTypes.TYPE_ERROR_DETAIL));
+                }
+                resourceFunctionPattern = replacePathPlaceHolders(resourceFunctionPattern, pathArgs);
+                MockRegistry.getInstance().registerCase(mockObj, resourceFunctionPattern, args, returnVal);
+            } else {
+                if (!validateReturnValue(functionName, returnVal, objectType.getMethods())) {
```

Comment:
let's invert and return early.

Reply:
I think here we do the same, we check the return value. If the return value does not match then, we return.